src/lib/entities/audio/model/audio-instance.ts

```diff
@@ -42,3 +43,29 @@ function clampTime(time: number) {
   const d = get(duration);
   return Math.max(0, Math.min(time, isNaN(d) ? 0 : d));
 }
+
+export function reportError(el: HTMLElement) {
+  // shameful workaround
+  if (!window.navigator.onLine) {
+    snackbar({ text: 'You are offline. Please connect to the internet and try again.' });
+    return;
+  }
+
+  const audioEl = el as HTMLAudioElement;
```

Comment:
A comment documenting the reason for the type assertion instead of using it as the parameter type would be helpful